pkg/piecestore/psserver/psdb/migrate_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package psdb_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+
+	"storj.io/storj/internal/dbutil/dbschema"
+	"storj.io/storj/internal/dbutil/sqliteutil"
+	"storj.io/storj/pkg/piecestore/psserver/psdb"
+)
+
+// loadSnapshots loads all the dbschemas from testdata/db.* caching the result
+func loadSnapshots() (*dbschema.Snapshots, error) {
+	snapshots := &dbschema.Snapshots{}
+
+	// snapshot represents clean DB state
+	snapshots.Add(&dbschema.Snapshot{
+		Version: -1,
+		Schema:  &dbschema.Schema{},
+		Script:  "",
+	})
+
+	// find all sql files
+	matches, err := filepath.Glob("testdata/db.*")
+	if err != nil {
+		return nil, err
+	}
+
+	for _, match := range matches {
+		versionStr := match[13 : len(match)-4] // hack to avoid trim issues with path differences in windows/linux
+		version, err := strconv.Atoi(versionStr)
+		if err != nil {
+			return nil, err
+		}
+
+		scriptData, err := ioutil.ReadFile(match)
+		if err != nil {
+			return nil, err
+		}
+
+		snapshot, err := sqliteutil.LoadSnapshotFromSQL(string(scriptData))
+		if err != nil {
+			return nil, err
+		}
+		snapshot.Version = version
+
+		snapshots.Add(snapshot)
+	}
+
+	snapshots.Sort()
+
+	return snapshots, nil
+}
+
+const newDataSeparator = `-- NEW DATA --`
+
+func newData(snap *dbschema.Snapshot) string {
+	tokens := strings.SplitN(snap.Script, newDataSeparator, 2)
+	if len(tokens) != 2 {
+		return ""
+	}
+	return tokens[1]
+}
+
+func TestMigrate(t *testing.T) {
+	snapshots, err := loadSnapshots()
+	require.NoError(t, err)
+
+	for _, base := range snapshots.List {
```

Comment:
We can have two base versions here only.

1. empty database
2. version 0

Rest of the migrations/versions will be implicitly tested by migrations themselves.

Reply:
I'm not sure if following, version 0 is from sqlite.v0.sql, empty db is added in `loadSnapshots` as -1

Reply:
It's more of adding a check for future, or in other words when someone adds version 2.

Reply:
ah analogy to https://github.com/storj/storj/blob/507fd817e4897a732d61b0e53edf38b1a45996b6/satellite/satellitedb/migrate_postgres_test.go#L102 ?

Reply:
yes

Reply:
done